admin/src/components/DownloadButton.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from "react";
+import LoadingButton from "./loadingButton";
+import styled from "styled-components";
+import * as FileSaver from "file-saver";
+import { toastr } from "react-redux-toastr";
+
+export default function DownloadButton({ source, title, ...rest }) {
+  const [buttonsLoading, setButtonsLoading] = useState(false);
+  return (
+    <DownloadBtn
+      {...rest}
+      color="white"
+      loading={buttonsLoading}
+      onClick={async () => {
+        const downloadAlert =
+          "En téléchargeant cette pièce jointe, vous vous engagez à la supprimer après consultation en application des dispositions légales sur la protection des données personnelles (RGPD, CNIL)";
+        if (confirm(downloadAlert)) {
+          setButtonsLoading(true);
+          try {
+            const f = await source();
+            const data = new Blob([new Uint8Array(f.data.data)], { type: f.mimeType });
+            FileSaver.saveAs(data, f.fileName.replace(/[^a-z0-9]/i, "-"));
```

Comment:
Pour sauver de la mémoire, peut être qu'on peut éviter de passer par une variable temporaire... Si ça se trouve le problème est là...
```suggestion
            FileSaver.saveAs(new Blob([new Uint8Array(f.data.data)], { type: f.mimeType }), f.fileName.replace(/[^a-z0-9]/i, "-"));
```